Enable PodMonitor for sharded mongo

francoisferrand opened the pull request:
This is required to get metrics.

Issue: ZENKO-3798

francoisferrand committed:
Enable PodMonitor for sharded mongo

This is required to get metrics.

Issue: ZENKO-3798

bert-e commented:
# Hello francoisferrand,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-3798 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.2.17`


Please check the `Fix Version/s` of ZENKO-3798, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-3798 contains:


* `2.2.14`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.2.17`


Please check the `Fix Version/s` of ZENKO-3798, or the target
branch of this pull request.

francoisferrand committed:
Remove alert from mongodb dashboard

This prevents grafana from properly loading the dashboard from
provisioning, and we use alertmanager for alerts anyway.

Issue: ZENKO-3798

francoisferrand committed:
Mongodb: remove datasource for rows

Issue: ZENKO-3798

francoisferrand committed:
Mongodb: fix rate range

Issue: ZENKO-3798

francoisferrand committed:
Mongodb: reorg oplog and resource panels

The last 2 sections used to be called "Alerts" and "Dashboard row", and
were also including some resource monitoring panels : best to just
consolidate everything in 2 rows dedicated to oplog (lag & size) and
resources (mem, disk, network, disk reads and disk writes).

Issue: ZENKO-3798

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand committed:
Mongodb: fix dashboard for sharded deployment

The metrics are the same, but are not split over multiple pods (mongos,
config-srv and shard-data)

Issue: ZENKO-3798

francoisferrand committed:
Mongodb: reorg dashboard for sharded deployment

Some metrics are available for each pod, some metrics for each
replicaset, and others for each mongod instance (i.e. subset of pods).

Thus introduce (hidden) "chained" variables and reorganise dashboard
panels to only show the panels for the relevant "objects".

Issue: ZENKO-3798

francoisferrand committed:
Mongodb: merge replicaset and oplog rows

The replicaset panels are actually computed per mongod pod, so it makes
more sense this way.

Issue: ZENKO-3798

francoisferrand committed:
Fix alerts for sharded mongodb

Issue: ZENKO-3798

francoisferrand commented:
/approve

bert-e commented:
# Conflict with a changeset in the queue



The changeset in this pull request conflicts with another changeset
already in the queue. Please wait for the current queue to merge into
the development branch. The conflict will then appear in this pull
request and can be sorted on the feature branch directly.

This changeset has *not* been added to the queue. 



*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.1`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.1`



Please check the status of the associated issue ZENKO-3798.

Goodbye francoisferrand.
